### ISSUE TITLE
FR: Display "Mail" under services

### ISSUE THREAD

**Issue by tyler36** [biati-digital/vscode-ddev-manager#2]
**Is your feature request related to a problem? Please describe.**
Thank you for the great extension, I've only been playing with for about 30 minutes but it's super!
I think it would be convienient to have a way to open the "Mail" service.

**Describe the solution you'd like**
I would like to see "Mail" under services, with an icon to open in the browser.
Clicking the icon, should open the current mail service in a browser, similar to `ddev launch -m` functionality.
Not sure if this is already there somewhere but I couldn't find it.

**Additional context**
DDEV `<=1.22.1` use Mailhog as the mail service.
There is an open [PR request](https://github.com/ddev/ddev/pull/5313) to replace MailHog with Mailpit; a different service with the same functionality and launch shortcut.
Perhaps call the service "Mail", instead of "MailHog" (current) or "Mailpit" (1.23 ?) would keep the name consistent regardless of version.

I am current running `ddev version v1.22.1-58-g3a427fd30` (PR preview version mentioned above).

**Comment by biati-digital** [comment]
Hi, I'm glad you like the extension.

That's already possible, in fact any service that provides a URL can be opened with the extension.

In the sidebar in the services list just move the mouse over mailhog and you'll see the action to launch it, other devices have different actions like PHP, MySQL, etc. you can also open mailhog using the command palette.

**Comment by biati-digital** [comment]
I read your message again, I think changing Mailhog or Mailpit to simply "Mail" it's a good idea, users moving from other tools will find it easier and know exactly what it is, at the end must of the users do not care if its Mailpit or mailhog, they just want to test emails.

I like it, I'll do the change.

**Comment by biati-digital** [comment]
mmmmm.... With your current version are you able to see the Mailhog service listed in the sidebar?

![CleanShot 2023-09-04 at 10 31 30](https://github.com/biati-digital/vscode-ddev-manager/assets/1219228/d795e7a4-676d-456c-9600-db29384c1e2d)
(I've changed the name to Mail)

I believe there's a bug with the current version of the extension, in my dev extension I see the Mail service but not in the extension published in the marketplace.

Anyway, I've made the change. I checked the PR you mention and added support for Mailpit, later when DDEV replaces Mailhog the extension will continue to work without problems. I'll publish an update soon.

**Comment by rfay** [comment]
Just FYI Mailhog/Mailpit aren't separate Docker "services". They're additional processes running inside the web container.

So is php-fpm

So is Node.js, but it's not normally "running" there

The database service is in fact a service, and if phpmyadmin is installed, it's a service.

**Comment by biati-digital** [comment]
Hi @rfay understood. I'll move NodeJS, PHP and Mail outside of services and display them with the other information. Just to be consistent on how DDEV works. Are you ok with using the label Mail instead of Mailhog or Mailpit?

![CleanShot 2023-09-04 at 11 10 12](https://github.com/biati-digital/vscode-ddev-manager/assets/1219228/ef2d7bc9-cefb-4204-9920-a2a61d11504c)

**Comment by rfay** [comment]
Maybe it would be better to use something like "Mail Catcher" or something? Not sure. It's actually the web UI that you're pointing to right?

**Comment by rfay** [comment]
I guess "Mail Catcher" isn't going to work; that's an alternate project. Just "Mail" should be OK for now.

**Comment by biati-digital** [comment]
Yes, clicking that will open the web UI for the selected project. I'll use "Mail" for now. 👍

**Comment by biati-digital** [comment]
@rfay  So "Mail Catcher" is out of the question? I think just "Mail" looks a little weird as the label of each item is getting shorter and all I see is that, not sure if it's just me. Or maybe using "Mail: Mailhog" or "Mail: Mailpit" or any other alternative like "SMTP Catcher"?

![CleanShot 2023-09-04 at 13 03 44](https://github.com/biati-digital/vscode-ddev-manager/assets/1219228/6035b0c1-1303-45a6-98ff-bf77fea6c852)

![CleanShot 2023-09-04 at 12 59 19](https://github.com/biati-digital/vscode-ddev-manager/assets/1219228/038aceba-1d02-4fff-8a15-15af94c5010d)
![CleanShot 2023-09-04 at 13 00 31](https://github.com/biati-digital/vscode-ddev-manager/assets/1219228/3080126c-7e96-4394-a271-8d05ae65b4ce)
![CleanShot 2023-09-04 at 13 09 40](https://github.com/biati-digital/vscode-ddev-manager/assets/1219228/19452427-313f-4307-b3ed-3bb6f1c1cd1d)

**Comment by rfay** [comment]
A lot of people don't know what "SMTP" means.  "Mail UI" ?

**Comment by tyler36** [comment]
> That's already possible, in fact any service that provides a URL can be opened with the extension.

On DDEV `1.22.1` I can see Mailhog, but do not DDEV `1.22.1-58-g3a427fd30`  (DDEV Manager `1.0.1`)

> Yes, clicking that will open the web UI for the selected project. I'll use "Mail" for now. 

"Mail UI" also works for me, if you think "Mail" is strange.

> I'll move NodeJS, PHP and Mail outside of services and display them with the other information. Just to be consistent on how DDEV works. 

I really like this. 👍

**Comment by biati-digital** [comment]
There's a new update that includes this and other changes.  [CHANGELOG](https://github.com/biati-digital/vscode-ddev-manager/blob/main/CHANGELOG.md)

About the "Mail" label, for now I'm using "Mail UI". I tend to be really picky about certain details of interfaces so I added a hidden setting just for myself to be able to change that text, I'll share it here just in case you want to try other alternatives:

```json
"ddevManager.sidebarLabels": {
    "email": "Mail Catcher",
  }
```

I've created a "[Roadmap](https://github.com/users/biati-digital/projects/1/views/1)"  just to share the things that I want to do/or remember that need to be added and also to share what I'm working on.
